### PR TITLE
Added the eval context manger which sets a module to evaluation mode

### LINKED ISSUE #18220: Context Manager that disables training mode with in a nn.Module.
## 🚀 Context Manager that disables training mode with in a nn.Module.
Currently `with torch.no_grad():` can be used to disable gradient calculation temporarily while batchnorm and drop out layers have to be set to evaluation mode using `model.train(True)` followed by a `model.train(False)` once evaluation is complete. Currently this syntax is used, 

```python
model.train(False)
with torch.no_grad():
  outputs = model(inputs)
  model.train(True)
  return outputs
```
. This is a potential source of bugs, as people will sometimes instead use `model.eval()` and forget to set the model back to training mode at the end of inference. Suggest having the following syntax, 

```python
with torch.no_train(model):
  with torch.no_grad():
    outputs = model(inputs):
    return outputs
```
. This syntax would stop this type of bug from occurring.

### PR DESCRIPTION
**Pull request by TamirCohen**
Fixes #18220
If you would think this feature is needed I will add tests

Usage:

```
import torch
lstm = torch.nn.LSTM(1, 200, 2, batch_first=True)

# The usage
with torch.nn.modules.module.eval(lstm):
    print(lstm.training)
```

### CODE REVIEW

**Comment by TamirCohen** [comment]
Got it, thanks for the response :)
